pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -865,14 +873,14 @@ func (r *ReconcileBareMetalHost) saveHostAnnotation(host *metal3v1alpha1.BareMet
 	}
 
 	if objStatus != nil {
-		// This value is copied to avoid continually updating the annotation
+		// These values are copied to avoid continually updating the annotation
 		objStatus.LastUpdated = host.Status.LastUpdated
+		objStatus.OperationHistory = host.Status.OperationHistory
 		if reflect.DeepEqual(host.Status, *objStatus) {
 			return nil
 		}
 	}
 
-	delete(host.Annotations, metal3v1alpha1.StatusAnnotation)
```

Comment:
shouldn't we delete the annotation first?

Reply:
no, it's replaced by the new one